wagtail/images/urls.py

```diff
@@ -1,7 +1,8 @@
-from django.conf.urls import url
+from django.urls import path
 
 from wagtail.images.views.serve import serve
 
 urlpatterns = [
-    url(r'^([^/]*)/(\d*)/([^/]*)/[^/]*$', serve, name='wagtailimages_serve'),
+    path('<str:signature>/<int:image_id>/<str:filter_spec>/', serve, name="wagtailimages_serve"),
+    path('<str:signature>/<int:image_id>/<str:filter_spec>/<str:filename>', serve, name="wagtailimages_serve")
```

Comment:
This line is here twice because of the optional `filename` part. I don't really like duplicating code this way but I thought I'd let this slip because the path converter syntax is so much clearer.

Reply:
Two URL routes pointing to the same view is fine, but are you sure it's valid to give them both the same value for `name`? Won't that cause `reverse` to fail?